.github/workflows/reusable-ecr.yml

```diff
@@ -0,0 +1,189 @@
+
+      - name: Build and push Docker images
+        uses: docker/build-push-action@v6
+        with:
+          context: .
+          file: ${{ matrix.file }}
+          load: ${{ matrix.image_name == 'frontend' && inputs.s3-upload == 'true' }}
+          push: ${{ env.SHOULD_PUSH_TO_ECR }}
+          tags: |
+            ${{ env.ECR_REPO }}/${{ matrix.image_name }}:${{ env.SHORT_SHA }}
+          platforms: linux/amd64
+          target: ${{ matrix.target }}
+          cache-from: type=local,src=/tmp/.buildx-cache
+          cache-to: type=local,dest=/tmp/.buildx-cache-new,mode=max
+
+      # Move cache to avoid unbounded cache growth
+      - name: Move cache
+        run: |
+          rm -rf /tmp/.buildx-cache
+          mv /tmp/.buildx-cache-new /tmp/.buildx-cache
+
+      - name: List Docker images
+        run: docker images
+
+      - name: Extract dist directory
+        if: ${{ matrix.image_name == 'frontend' && inputs.s3-upload == 'true' }}
+        run: |
+          IMAGE_TAG="${{ env.ECR_REPO }}/${{ matrix.image_name }}:${{ env.SHORT_SHA }}"
+          echo "Using image: $IMAGE_TAG"
+          docker image inspect $IMAGE_TAG
+          CONTAINER_ID=$(docker create $IMAGE_TAG)
+          echo "Created container: $CONTAINER_ID"
+          mkdir -p /tmp/extracted-dist
+          docker cp $CONTAINER_ID:/app/client/dist /tmp/extracted-dist/
+          ls -al /tmp/extracted-dist
+          docker rm $CONTAINER_ID
+          echo "Extracted files:"
+          ls -la /tmp/extracted-dist/dist/
+          echo "Successfully extracted dist directory from the frontend image"
+
+      - name: Upload dist as artifact
+        if: ${{ matrix.image_name == 'frontend' && inputs.s3-upload == 'true' }}
+        uses: actions/upload-artifact@v4
+        with:
+          name: frontend-dist
+          path: /tmp/extracted-dist/dist/
+          retention-days: 1
+
+  upload-config:
+    name: Upload styling to S3
+    needs: build
+    if: ${{ inputs.s3-upload == 'true' }}
+    runs-on: ubuntu-latest
+    env:
+      SHARED_CONFIG_BUCKET: genai-shared-config
+    permissions:
+      id-token: write # Required for AWS authentication
+      contents: read
+    steps:
+      - name: Download dist artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: frontend-dist
+          path: /tmp/dist
+
+      - name: Zip dist directory
+        run: |
+          cd /tmp/dist
+          zip -r frontend-dist.zip .
+          echo "Created /tmp/dist/frontend-dist.zip"
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: arn:aws:iam::${{ env.AWS_ACCOUNT_ID }}:role/github-oidc-genai
+          aws-region: ${{ env.AWS_REGION }}
+
+      - name: Upload to S3
+        run: |
+          aws s3 cp /tmp/dist/frontend-dist.zip s3://${{ env.SHARED_CONFIG_BUCKET }}/frontend/dist/${{ github.sha }}/dist.zip"
```

Comment:
The only nitpick I have is if we could use short_sha here. But it's not the highest priority.

Reply:
I think we should not risk future failures when short sha already exists